src/Cache/MongoStore.php

```diff
@@ -0,0 +1,282 @@
+    {
+        return new MongoLock(
+            ($this->lockConnection ?? $this->connection)->getCollection($this->lockCollectionName),
+            $this->prefix . $name,
+            $seconds,
+            $owner,
+            $this->lockLottery,
+            $this->defaultLockTimeoutInSeconds,
+        );
+    }
+
+    /**
+     * Restore a lock instance using the owner identifier.
+     */
+    #[Override]
+    public function restoreLock($name, $owner): MongoLock
+    {
+        return $this->lock($name, 0, $owner);
+    }
+
+    /**
+     * Store an item in the cache for a given number of seconds.
+     *
+     * @param  string $key
+     * @param  mixed  $value
+     * @param  int    $seconds
+     */
+    #[Override]
+    public function put($key, $value, $seconds): bool
+    {
+        $result = $this->collection->updateOne(
+            [
+                'key' => $this->prefix . $key,
+            ],
+            [
+                '$set' => [
+                    'value' => $this->serialize($value),
+                    'expiration' => $this->currentTime() + $seconds,
+                ],
+            ],
+            [
+                'upsert' => true,
+
+            ],
+        );
+
+        return $result->getUpsertedCount() > 0 || $result->getModifiedCount() > 0;
+    }
+
+    /**
+     * Store an item in the cache if the key doesn't exist.
+     *
+     * @param  string $key
+     * @param  mixed  $value
+     * @param  int    $seconds
+     */
+    public function add($key, $value, $seconds): bool
+    {
+        $result = $this->collection->updateOne(
+            [
+                'key' => $this->prefix . $key,
+            ],
+            [
+                [
+                    '$set' => [
+                        'value' => [
+                            '$cond' => [
+                                'if' => ['$lte' => ['$expiration', $this->currentTime()]],
+                                'then' => $this->serialize($value),
+                                'else' => '$value',
+                            ],
+                        ],
+                        'expiration' => [
+                            '$cond' => [
+                                'if' => ['$lte' => ['$expiration', $this->currentTime()]],
+                                'then' => $this->currentTime() + $seconds,
+                                'else' => '$expiration',
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+            ['upsert' => true],
+        );
+
+        return $result->getUpsertedCount() > 0 || $result->getModifiedCount() > 0;
+    }
+
+    /**
+     * Retrieve an item from the cache by key.
+     *
+     * @param  string $key
+     */
+    #[Override]
+    public function get($key): mixed
+    {
+        $result = $this->collection->findOne(
+            [
+                'key' => $this->prefix . $key,
```

Comment:
Did you consider using `_id` for the cache key field? I assume there shouldn't be duplicates, and this would ensure that the queries are indexed.